src/lib/__tests__/AnnotationControlsFSM-test.js

```diff
@@ -0,0 +1,90 @@
+import AnnotationControlsFSM, { AnnotationInput, AnnotationState } from '../AnnotationControlsFSM';
+import { AnnotationMode } from '../AnnotationControls';
+
+let annotationControlsFSM;
+
+describe('lib/AnnotationControlsFSM', () => {
+    beforeEach(() => {
+        annotationControlsFSM = new AnnotationControlsFSM();
+    });
+
+    [
+        {
+            state: AnnotationState.NONE,
```

Comment:
Is this all the possible state changes? Maybe consider a test for each state and test the transition (if any) based on all possible inputs?

Reply:
The FSM graph is symmetry, so I think it's enough to only test upper/lower half of the states and transitions.

Reply:
It makes sense to group tests by state. Will do.